app/validators/user_role_name_validator.rb

```diff
@@ -0,0 +1,9 @@
+class UserRoleNameValidator < ActiveModel::Validator
```

Comment:
I didn't like doing it because it's there purely so specs would pass. There's some detail in the commit comment f283575b535de77056dd9b386d8937326ff713d0 but it came down to `UserRole.valid_roles` being "frozen" at load time, so enabling the `billing_administrator` feature in a context wouldn't necessarily add `billing_administrator` to the list as I'd expect.